test/memspaces/mempolicy.cpp

```diff
@@ -109,5 +112,63 @@ TEST_F(test, mempolicyInterleavePartSize) {
     EXPECT_EQ(ProviderInternal->numa_flags,
               HWLOC_MEMBIND_BYNODESET | HWLOC_MEMBIND_STRICT);
     EXPECT_EQ(ProviderInternal->part_size, part_size);
+    EXPECT_EQ(ProviderInternal->mode, UMF_NUMA_MODE_INTERLEAVE);
+    umfMemoryProviderDestroy(hProvider);
+}
+
+TEST_F(test, mempolicyDefaultSplit) {
+    umf_memory_provider_handle_t hProvider = nullptr;
+    umf_mempolicy_handle_t hPolicy = nullptr;
+
+    umf_result_t ret = umfMempolicyCreate(UMF_MEMPOLICY_SPLIT, &hPolicy);
+    ASSERT_EQ(ret, UMF_RESULT_SUCCESS);
+
+    ret = umfMemoryProviderCreateFromMemspace(umfMemspaceHostAllGet(), hPolicy,
+                                              &hProvider);
+    ASSERT_EQ(ret, UMF_RESULT_SUCCESS);
+    ASSERT_NE(hProvider, nullptr);
+    ret = umfMempolicyDestroy(hPolicy);
+    ASSERT_EQ(ret, UMF_RESULT_SUCCESS);
+
+    os_memory_provider_t *ProviderInternal =
+        (os_memory_provider_t *)providerGetPriv(hProvider);
+    ASSERT_NE(ProviderInternal, nullptr);
+    EXPECT_EQ(ProviderInternal->numa_policy, HWLOC_MEMBIND_BIND);
+    EXPECT_EQ(ProviderInternal->numa_flags,
+              HWLOC_MEMBIND_BYNODESET | HWLOC_MEMBIND_STRICT);
+    EXPECT_EQ(ProviderInternal->partitions_len, ProviderInternal->nodeset_len);
+    EXPECT_EQ(ProviderInternal->mode, UMF_NUMA_MODE_SPLIT);
+    umfMemoryProviderDestroy(hProvider);
+}
+
+TEST_F(test, mempolicyCustomSplit) {
+    umf_memory_provider_handle_t hProvider = nullptr;
+    umf_mempolicy_handle_t hPolicy = nullptr;
+
+    umf_result_t ret = umfMempolicyCreate(UMF_MEMPOLICY_SPLIT, &hPolicy);
+    ASSERT_EQ(ret, UMF_RESULT_SUCCESS);
+    umf_mempolicy_split_partition_t part[] = {{1, 0}, {1, 0}};
```

Comment:
shouldn't we test more various patterns (similarly to your prev. PR)? at best with randomization...?

Reply:
my idea is that os_provider is allready tested - so here i only test copy one struct do different one - i do not want to copy all numa tests just do change how we create osprovider. This is why tests in this file are quite simple